### ISSUE TITLE
Cannot use ES6 Arrow functions with Robomongo 0.9.0

### ISSUE THREAD

**Issue by nkahnfr** [Studio3T/robomongo#1197]
Hi,

I didn't find a way to get following code running in robomongo shell.
`db.collection.find({}).limit(5).map((doc) => { return doc._id; });`
- Expected result: an array containing 5 identifiers.

```
[
  "00044ca2c34c3972ec2c3845e9691cd2",
  "0001bda6051f4286720717e0bbed34c5",
  "000829c6b7f411d6658e65b91a4278d2",
  "000506f36ccc90b6c23370cfb6698abc",
  "0003e5151e3deab2a9080c64d23a8dde"
]
```
- Actual result: an error message.

```
Error: Line 1: Unexpected token >
```

The same code works fine using mongo interactive JavaScript shell.
`mongo my-db --eval "db.collection.find({}).limit(5).map((doc) => { return doc._id; });"`

I am using latest version of Robomongo (0.9.0-RC10) on a MacBook Pro (OS X 10.11.6).
Is there any configuration required to get ES6 working in robomongo shell?

Thanks for your help.

-Nicolas

UPD: adding details on expected and actual results.

**Comment by simsekgokhan** [comment]
Hi @nkahnfr , thanks a lot for reporting the problem. We have reproduced the problem. We have put this issue into our list to include in next releases.

For now, normal syntax can be used. I assume you are already achieving the same result using the syntax below?

`db.collection.find({}).limit(5).map(function(doc) { return doc._id; });
`

**Comment by nkahnfr** [comment]
Hi @simsekgokhan , thanks for your response.

Indeed, I confirm that standard function expressions are working correctly.

Somehow I came to issue #977 and was curious to test ES6 support that should be provided by the SpiderMonkey engine.

**Comment by sudowork** [comment]
Hi @simsekgokhan , I'm wondering if this is still on the roadmap. As of Robomongo 1.0, arrow function expressions still do not appear to be supported. Thanks!

**Comment by simsekgokhan** [comment]
Hi @sudowork , @nkahnfr ,
We did some investigation and seems like we identified the initial changes to support ES6 features where arrow functions and others are usable. Currently, ES6 support is planned to be in next release with 3.4 support.

However it is possible to provide a beta version if you guys do not want to wait and test the beta.
Please share your OS if you are interested in this beta.

**Comment by simsekgokhan** [comment]
@sudowork , @nkahnfr 
Please see the beta which includes ECMAScript 2015 (ES6):  [Robomongo 1.1 - Beta version with MongoDB 3.4 Support](https://github.com/Studio3T/robomongo/issues/1250#issuecomment-301036770)

**Comment by sudowork** [comment]
Thanks so much @simsekgokhan!! Really appreciate the work on this :).

**Comment by ccpalmer** [comment]
Still cannot connect to the free Atlas cluster.  I get this:
![2017-05-16_23-29-14](https://cloud.githubusercontent.com/assets/4670554/26137354/a7ccbf0a-3a8f-11e7-9aed-a6deecfa6a32.png)
Password is correcand SSL Self-signed is selected.

any thots?

**Comment by simsekgokhan** [comment]
@ccpalmer 
Can you confirm you are using Robomongo 1.1-Beta?
Please try with entering your replica set's set name into Robomongo connection settings?
And, make sure all of the member names of your Mongo Atlas cluster are correctly entered in Robomongo connection settings. Please let us know the outcome.

**Comment by ccpalmer** [comment]
Duh ...  u are right.  I was using 1.0  Thank you!!!

**Comment by juliashibalko** [comment]
Fixed and verified in Robo3T 1.1 version. Closing as fixed.
Please feel free to reopen in any case.
